### PR TITLE
[cni-cilium] Improved PodMonitors

### PR DESCRIPTION
**Pull request by zuzzas**
## Description

Naming and apply them only if operator-prometheus-crd is enabled.

## Why do we need it, and what problem does it solve?

PodMonitors were not appropriately named. This resulted in non-transparent understanding of metrics labels and their source.

## What is the expected result?

Metrics are now appropriately named.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cni-cilium
type: fix
summary: Improved PodMonitors. Naming and apply them only if operator-prometheus-crd is enabled.
impact_level: low
```

### CODE REVIEW

**Comment by zuzzas** [comment]
It is displayed like this now. I believe that should be enough, since the namespace label clearly indicates `d8-cni-cilium`. Changing `jobLabel` will trigger global cilium restart, since I'd have to change Pod labels as well.

image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/021-cni-cilium/templates/agent/podmonitor.yaml [hunk 1]
```diff
@@ -1,8 +1,9 @@
+{{- if (.Values.global.enabledModules | has "operator-prometheus-crd") }}
 ---
 apiVersion: monitoring.coreos.com/v1
 kind: PodMonitor
 metadata:
-  name: agent
+  name: cilium-agent
   namespace: d8-monitoring
   {{- include "helm_lib_module_labels" (list . (dict "app" "agent" "prometheus" "main")) | nindent 2 }}
 spec:
```
Comment (nabokihms) [inline modules/021-cni-cilium/templates/agent/podmonitor.yaml:9]: Do you mind also changing the job label value?

**Review by nabokihms — APPROVED** [review]

**Review by nabokihms — CHANGES_REQUESTED** [review]
Job label values are agent and operator. Consider changing these two also.

**Comment by zuzzas** [comment]
@nabokihms But I can't do it easily. `jobLabel` equals `app` now. That means that it will take the `app` key from Pod labels. And I'll have to change them all for this to work.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/021-cni-cilium/templates/agent/podmonitor.yaml (hunk 1 above)
Comment (zuzzas) [reply modules/021-cni-cilium/templates/agent/podmonitor.yaml:9]: Do I change every single label in all resources? To be uniform. But that will rolling-restart cilium pods.

File: modules/021-cni-cilium/templates/agent/podmonitor.yaml [hunk 2]
```diff
@@ -19,9 +20,12 @@ spec:
     relabelings:
     - targetLabel: tier
       replacement: cluster
+    - targetLabel: job
+      replacement: "cilium-operator"
```
Comment (nabokihms) [inline modules/021-cni-cilium/templates/agent/podmonitor.yaml:24]:
```suggestion
      replacement: "cilium-agent"
```

**Review by nabokihms — APPROVED** [review]

**Pull request merged**